x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -317,3 +329,50 @@ export const getAgentDataHandler: RequestHandler<
 function isStringArray(arr: unknown | string[]): arr is string[] {
   return Array.isArray(arr) && arr.every((p) => typeof p === 'string');
 }
+
+interface VersionsType {
+  versions: string[];
+}
+
+// Read a static file generated at build time and cache it for subsequent calls
+export const getAvailableVersionsHandler: RequestHandler = async (context, request, response) => {
+  const versionsFile = 'build/kibana/x-pack/plugins/fleet/target/agent_versions_list.json';
```

Comment:
I share a similar concern to @jbudz here, the `build/kibana` directory is just the temporary directory used after running the build locally and it seems improper for the fleet plugin to be expecting it to exist.

Reply:
The correct path should be `x-pack/plugins/fleet/target/agent_versions_list.json` correct?

Reply:
That would be a fine place to store things, the `target` directory is gitignored.

Reply:
Thanks, I wasn't aware that the `build/kibana` directory is not available. TIL! :)